File: backend/app/schemas/digital_id.py
"""Pydantic schemas for the digital-ID / QR verification feature."""
import datetime

from pydantic import BaseModel


class DigitalIdRequest(BaseModel):
    """Payload for POST /api/verify-id."""

    token: str


class DigitalIdResponse(BaseModel):
    """Response body returned by POST /api/digital-id/{tourist_id}."""

    tourist_id: int
    token: str
    expires_at: datetime.datetime
    qr_code: str


class VerifiedTourist(BaseModel):
    """Minimal tourist info returned after a successful verification."""

    id: int
    name: str


class VerifyResponse(BaseModel):
    """Response body returned by POST /api/verify-id."""

    valid: bool
    tourist: VerifiedTourist | None = None

File: frontend/src/api/digitalId.ts
/** Digital-ID generation & verification API module. */
import { apiPost } from "./client";
import type {
  DigitalIdResponse,
  VerificationResponse,
  VerifyDigitalIdPayload,
} from "../types/tourist";

/** POST /api/digital-id/{tourist_id} */
export function createDigitalId(touristId: number): Promise<DigitalIdResponse> {
  return apiPost<DigitalIdResponse>(`/api/digital-id/${touristId}`);
}

/** POST /api/verify-id */
export function verifyDigitalId(payload: VerifyDigitalIdPayload): Promise<VerificationResponse> {
  return apiPost<VerificationResponse>("/api/verify-id", payload);
}
